dashboard/app/controllers/test_controller.rb

```diff
@@ -142,9 +142,13 @@ def assign_course_as_student
     end
 
     section_name = params[:section_name] || "New Section"
-    section = Section.create(name: section_name, user: teacher_user, course_id: course.id, participant_type: Curriculum::SharedCourseConstants::PARTICIPANT_AUDIENCE.student)
+    unit = course.single_unit_course? ? course.default_unit_group_units.first.script : nil
+    section = Section.create!(name: section_name, user: teacher_user, course_id: course.id, script: unit, participant_type: Curriculum::SharedCourseConstants::PARTICIPANT_AUDIENCE.student)
     section.students << user
     section.save!
+
+    UserScript.create!(user: user, script: unit, assigned_at: Time.now) if unit
```

Comment:
this turned out not to be strictly necessary to make UI tests pass, but brings our UI test assignment code to more closely match what changes in the DB when we assign courses via the UI

Reply:
Similar to my comment in the other PR: could we use the same API as the UI ?